mmf/utils/distributed.py

```diff
@@ -169,6 +196,11 @@ def byte_tensor_to_object(byte_tensor, max_size=4094):
 def infer_init_method(config):
     if config.distributed.init_method is not None:
         return
+
+    if getattr(config.training, 'device', 'cuda') == 'xla':
+        global _USE_XLA
```

Comment:
I would suggest that we use MMF's registry to track globals.

Reply:
I changed to use the registry. It resulted into circular reference:
``` File "/home/sivaibhav/mmf/mmf_cli/run.py", line 9, in <module>
    from mmf.common.registry import registry
  File "/home/sivaibhav/mmf/mmf/__init__.py", line 5, in <module>
    from mmf import utils, common, modules, datasets, models
  File "/home/sivaibhav/mmf/mmf/common/__init__.py", line 3, in <module>
    from .registry import registry
  File "/home/sivaibhav/mmf/mmf/common/registry.py", line 27, in <module>
    from mmf.utils.env import setup_imports
  File "/home/sivaibhav/mmf/mmf/utils/env.py", line 13, in <module>
    from mmf.utils.general import get_absolute_path
  File "/home/sivaibhav/mmf/mmf/utils/general.py", line 10, in <module>
    from mmf.utils.distributed import get_rank, get_world_size
  File "/home/sivaibhav/mmf/mmf/utils/distributed.py", line 13, in <module>
    from mmf.common.registry import registry
ImportError: cannot import name 'registry'
```
I attempted to break the circular import by pushing get_absolute_path into the function where it was used. Please review.